lib/spack/spack/test/patch.py

```diff
@@ -18,20 +19,26 @@
 from spack.spec import Spec
 from spack.stage import Stage
 from spack.util.executable import Executable
+from spack.util.path import is_windows
 
 # various sha256 sums (using variables for legibility)
+# many file based shas will differ between Windows and other platforms
+# due to the use of carriage returns ('\r\n') in Windows line endings
```

Comment:
If the git repo checks out with Unix-style line endings, would that avoid the hash differences? If so, then I think we could specify that in the CI. Or would that create another issue (i.e. if tools can't work with Unix-style line endings on Windows). If this works for the unit tests but not in general than IMO it would be find to leave the hashes as is.

Reply:
I have that setting locally for all projects, and it seems to have no impact here. Either way, I'd rather not default to that, I'm not aware of any tools that will break without carriage returns, it may result in some bizarre errors down the way. I don't like keeping track of two different hashes, but for the test I think its alright for now. I've had no issues with patches thus far however actually using spack.